subctl/issue-182-subctl-a-kubectl-plugin.md

```diff
@@ -0,0 +1,149 @@
+# Support krew way of installing `subctl`
+
+Related Issue:
+[Add subctl as kubectl plugin](https://github.com/submariner-io/enhancements/issues/182)
+
+## Background
+
+> Krew is a tool that makes it easy to use kubectl plugins. Krew helps you discover plugins, install and manage them on your machine. It
+> is similar to tools like apt, dnf or brew. Today, over [200 kubectl plugins](https://krew.sigs.k8s.io/plugins/) are available on Krew.
+>
+> A plugin is a standalone executable file, whose name begins with `kubectl-`.
+>
+> More information can be found on [extending kubectl with plugins](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/)
+> page.
+>
+> On the surface, installing a kubectl plugin seems simple enough – all you need to do is to place an executable in the user’s `PATH`
+> prefixed with `kubectl-` – that you may be considering some other alternatives to Krew, such as:
+>
+> * Having the user manually download the plugin executable and move it to some directory in the PATH
+> * Distributing the plugin executable using an OS package manager, like Homebrew (macOS), apt/yum (Linux), or Chocolatey (Windows)
+> * Distributing the plugin executable using a language package manager (e.g. npm or go get)
+>
+> While these approaches are not necessarily unworkable, potential drawbacks to consider include:
+>
+> * How to get updates to users (in the case of manual installation)
+> * How to package a plugin for multiple platforms (macOS, Linux, and Windows)
+> * How to ensure your users have the appropriate language package manager (go, npm)
+> * How to handle a change to the implementation language (e.g. a move from npm to another package manager)
+>
+> Krew solves these problems cleanly for all kubectl plugins, since it’s designed specifically to address these shortcomings. With Krew,
+> after you write a plugin manifest once your plugin can be installed on all platforms without having to deal with their package managers.
+
+## Summary
+
+krew is a kubectl plugin that makes lifecycle management of kubectl plugins easy. The idea is to have `subctl` as a kubectl plugin and be
```

Comment:
nitpick: Shouldn't krew be `krew`?

Reply:
It should be Krew (see [the website](https://krew.sigs.k8s.io/)).

Reply:
yes, that's also one of the golas: to make subctl (built with build constraints)  available downstream via oc as part of OpenShift CLI manager.